charts/opentelemetry-demo/values.yaml

```diff
@@ -39,19 +38,28 @@ serviceAccount: ""
 components:
   redis:
     enabled: true
+    default:
+      env:
+        enabled: true
```

Comment:
Am I understanding it right that, if I want to have other set of envs for a particular component, I need to:
```yaml
components:
  redis:
    default:
      env:
        enabled: false
      values:
        - name: CUSTOM_ENV
          value: custom_val 
```

I don't think this is an intuitive interface for users... 

Maybe something like?

```yaml
components:
  redis:
    useDefault:
      env: false
    env:
      - name: CUSTOM_ENV
        value: custom_val 
```

Reply:
@dmitryax the latter is what is proposed in this PR. I stuck with our standard "env.enabled" syntax but I am ok combining "enabled" and "env" into one.

Reply:
> @dmitryax the latter is what is proposed in this PR. 

Not sure I understand what interface this PR provides then. Can you please post a snippet?

What I understood (the first section) just doesn't seem right. For example why `components.default.values` is used to override default `env`?

Reply:
I'll provide more clarification when I'm back from PTO on the 13th.

Reply:
@dmitryax done.